### PR TITLE
Implemented Check-Id in ftd and tested this using a hard coded dummy id map

### PR DESCRIPTION
**Pull request by Heulitig**
Discussion: [globally unique id](https://github.com/FifthTry/fpm/discussions/450)

### CODE REVIEW

**Comment by Heulitig** [comment]
@Arpita-Jaiswal Im done with this PR can you merge this after reviewing it ?

**Comment by Arpita-Jaiswal** [comment]
@Heulitig Please run `cargo fmt`.

File: src/p2/interpreter.rs [hunk 1]
```diff
@@ -736,6 +766,126 @@ impl ParsedDocument {
...
+                dbg!(key, value);
+            }
+
+            dbg!(&section.body);
+            for subsection in section.sub_sections.0.iter() {
+                dbg!(
+                    &subsection.name,
+                    &subsection.caption,
+                    &subsection.is_commented
+                );
```
Comment (AbrarNitk) [inline src/p2/interpreter.rs:878]: @Heulitig Please remove `dbg!` statements.

**Review by AbrarNitk — CHANGES_REQUESTED** [review]

File: src/p2/interpreter.rs [hunk 2]
```diff
@@ -736,6 +766,117 @@ impl ParsedDocument {
...
+
+            // Syntax 1 = [id`](id: someId)
+            // G0 = Original capture, G1 = Linked Text, G2 = Actual Id
+            for cap in S1.captures_iter(text) {
+                let replacement = format!(
+                    "[{}]({}#{})",
+                    capture_group_at(&cap, 1).trim(),
+                    &global_ids[capture_group_at(&cap, 2).trim()],
+                    slug::slugify(capture_group_at(&cap, 2).trim())
+                );
```
Comment (Arpita-Jaiswal) [inline src/p2/interpreter.rs:833]:
```suggestion
                let linked_text =  capture_group_at(&cap, 1).trim();
                let actual_id = capture_group_at(&cap, 2).trim();
                let link = global_ids.get(actual_id).ok_or_else(<something>) // throw error if not found
                let replacement = format!(
                    "[{}]({}#{})", linked_text, link, slug::slugify(actual_id)
                );
```

File: src/p2/interpreter.rs [hunk 3]
```diff
@@ -736,6 +766,117 @@ impl ParsedDocument {
...
+            // Syntax 2 = {id: someId}
+            // G0 = Original capture, G1 = Actual Id
+            for cap in S2.captures_iter(text) {
+                let replacement = format!(
+                    "[{}]({}#{})",
+                    capture_group_at(&cap, 1).trim(),
+                    &global_ids[capture_group_at(&cap, 1).trim()],
+                    slug::slugify(capture_group_at(&cap, 1))
+                );
+                result = result.replacen(capture_group_at(&cap, 0), &replacement, 1);
```
Comment (Arpita-Jaiswal) [inline src/p2/interpreter.rs:846]: Same as above

File: src/p2/interpreter.rs [hunk 4]
```diff
@@ -736,6 +766,117 @@ impl ParsedDocument {
...
+
+            result
+        }
+    }
+
+    /// prints the parsed document (for debugging purposes)
+    #[allow(dead_code)]
+    fn show_document(&self) {
+        for section in self.sections.iter() {
+            dbg!(&section.name, &section.caption, section.is_commented);
```
Comment (Arpita-Jaiswal) [inline src/p2/interpreter.rs:857]: Remove Logs

**Review by Arpita-Jaiswal — CHANGES_REQUESTED** [review]

**Comment by Heulitig** [comment]
@Arpita-Jaiswal 
Done with the changes do review this again

**Pull request merged**